### PR TITLE
Add junos vlans resource module

### PR DESCRIPTION
**Pull request by danielmellado**
##### SUMMARY
Add junos_vlans

This commit adds junos_vlans resource module, allowing operations such as these
for l3 interfaces configuration in junos

- merge
- delete
- override
- replace

The bootstrapping from this module has been done using resource module builder
[1]

[1] https://github.com/ansible-network/resource_module_builder

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
junos_vlans

##### ADDITIONAL INFORMATION
As this module uses netconf, it does require ncclient, as mentioned there.

Depends-On: https://github.com/ansible/ansible-zuul-jobs/pull/168

### CODE REVIEW

**Comment by pabelanger** [comment]
recheck

File: lib/ansible/module_utils/network/junos/config/vlans/vlans.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        """
+        intf_xml = []
+
+        if not want:
+            want = have
+
+        for config in want:
+            vlan_name = config['name']
+            vlan_root = build_root_xml_node('vlan')
+            vlan_root.attrib.update({'delete': 'delete'})
```
Comment (trishnaguha) [inline lib/ansible/module_utils/network/junos/config/vlans/vlans.py:205]: @danielmellado Does `state_deleted` delete the vlan or remove configurations from the vlan when `config` is given in a task?

**Comment by danielmellado** [comment]
recheck

File: lib/ansible/module_utils/network/junos/config/vlans/vlans.py (hunk 1 above)
Comment (danielmellado) [reply lib/ansible/module_utils/network/junos/config/vlans/vlans.py:205]: it does delete the vlan, you can check what would happen in the deleted sample ;)

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now fde1318295456d3fc714a38a0d1c9327ca9efa34

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 46b677782ef2d90205af55555b95e6a5b5d162a3

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now c870b7994ccba53af42dee331d50ab6e2de24616

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 2bf022e3d5c5f84c1369ecf63d2d4e5bdd034b39

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now f76e6bab2528ab36abed4eb17a7b3f7fff7ac8a2

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 10e99ebb11394ac1f34fdcd56f777bb6e793561d

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 99d44b0043d932982843e27888c59ad55f4fd7fa

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 9e565e7ea39b9235599abe5bc7adb3513f5a2db1

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 4c5f1182e98f9873b5a3c3df5df09790a4dcc93a

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 2a778909a73fea97103953706f788ff76564d3d2

File: lib/ansible/modules/network/junos/junos_vlans.py [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+  returned: when changed
+  type: str
+  sample: >
+    The configuration returned will always be in the same format
+     of the parameters above.
+commands:
+  description: The set of commands pushed to the remote device.
+  returned: always
+  type: list
+  sample: ['command 1', 'command 2', 'command 3']
```
Comment (ganeshrn) [inline lib/ansible/modules/network/junos/junos_vlans.py]: since xml list is returned this can be modified to `sample: ['xml 1', 'xml 2', 'xml 3']`

**Review by ganeshrn — APPROVED** [review]
Apart from the `return` comment the PR looks good to me

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 1abb67d09e1f917e607a25b04eb5aadee0f1ca88

**Comment by danielmellado** [comment]
recheck

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 6d20a9f883a141d3e1d3950851f59e719d5f3c46

**Comment by danielmellado** [comment]
recheck

**Pull request merged**